public/app/features/explore/ContentOutline/ContentOutlineItem.tsx

```diff
@@ -25,7 +27,7 @@ export function ContentOutlineItem({ title, icon, children, className }: Content
   }, [title, icon, register, unregister]);
 
   return (
-    <div className={className} ref={ref}>
+    <div className={className} ref={ref} data-testid={selectors.components.Panels.Panel.title(title)}>
```

Comment:
We use PanelChrome for panels so the testid is already provided in there. Table panel may have dynamic suffix but still can be selected with a substring matching like `document.querySelector('*[data-testid^="data-testid Panel header Table"')`. We don't guarantee panel title won't change in the future but that's the same case for ContentOutlineItem.title (they basically should match). 

On top of that ContentOutlineItem.tsx is not used to wrap panels but items included in the outline. It includes panels but also other elements like query rows and more stuff in the future. Having `data-testid Panel header ...` on non-panel elements may cause problems selecting the correct element.

If you need something stable that won't change from version to version we may need to introduce something hardcoded, not based on UI elements.

Reply:
I see - it makes sense! 

I took a closer look at the code, and the problem (from an e2e testing perspective) is that _sometimes_ the selector for the table panel in Explore resolves to `data-testid="data-testid Panel header Table"` (this is good), whereas for some data source responses it resolves to `data-testid="Panel"` (not good). 

The `Panel` selector comes from [PanelChrome](https://github.com/grafana/grafana/blob/815e61258c2cd755440a0d64a33feb005153d4d7/packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx#L164), but the problem appears to be in the [TableContainer](https://github.com/grafana/grafana/blob/main/public/app/features/explore/Table/TableContainer.tsx#L56-L67) and related to translations. I'm only seeing the bad selector when the DataFrame doesn't have a name, and this seems to be fairly common I'm afraid. 

It would be good if the PanelChrome component allowed consumer to pass a `dataTestIdSuffix` instead of relying on the title which can be a React element. But I'm not too familiar with this part of the code base. Maybe you have a better idea?

Reply:
O I see! In that case we can pass a translated string, not a react node:

```diff
-      <Trans i18nKey="explore.table.title-with-name">Table - {{ name }}</Trans>
+      t('explore.table.title-with-name', 'Table - {{name}}', { name })
```

That should make it work:

<img width="708" alt="Screenshot 2024-02-15 at 11 36 40" src="https://github.com/grafana/grafana/assets/745532/9768e0fa-885f-49a2-8f02-a4e875d390e6">



It's also a bit unexpected how PanelChrome generated a test id when a non-string is passed. @grafana/grafana-frontend-platform any reason why it's not prefixed in a consistent way, e.g. with something like:

```diff
-  const testid = typeof title === 'string' ? selectors.components.Panels.Panel.title(title) : 'Panel';
+  const testid = typeof title === 'string' ? selectors.components.Panels.Panel.title(title) : selectors.components.Panels.Panel.title('Unknown')
```

Reply:
@sunker here's a branch with proposed change: https://github.com/grafana/grafana/compare/ifrost/translate-table-title-in-runtime?expand=1. Would that work?

Reply:
@ifrost I took that branch for a spin, and it solves that problem! 🥇  Do you want to raise the PR? You can assign me as reviewer if you want.

Reply:
https://github.com/grafana/grafana/pull/83236